src/components/dialog/dialog.js

```diff
@@ -151,45 +152,15 @@ class Dialog extends Modal {
    */
   get dialogTitle() {
     if (this.props.title) {
-      return <h2 className={ this.dialogTitleClasses } data-element='title'>{ this.props.title }</h2>;
-    }
-
-    return null;
-  }
-
-  /**
-   * Returns HTML and text for the dialog subtitle.
-   *
-   * @method dialogSubtitle
-   * @return {String} subtitle to display
-   */
-  get dialogSubtitle() {
-    if (this.props.subtitle) {
-      return <p className={ this.dialogSubtitleClasses } data-element='subtitle'>{ this.props.subtitle }</p>;
+      return (
+        <Heading title={ this.props.title } subheader={ this.props.subtitle } />
```

Comment:
We'll need to be able to set a `id` attribute on the `Heading` `title` and `subtitle`. This can then be used to add `aria-labelledby` and `aria-describedby` attributes to the `Dialog` itself, which will help the dialog announce itself to screen readers, etc.

For example:

```
<Heading
  title={ this.props.title}
  titleId={ this.props.titleId }
  subtitle={ this.props.subtitle }
  subtitleId={ this.props.subtitleId }
/>

// The when you render the dialog
<div 
  role="dialog" 
  aria-labelledby={ this.props.titleId } 
  aria-describedby={ this.props.subtitleId }
>
  ...
</div>
```